vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/test/grid-connector-tree.test.ts

```diff
@@ -89,12 +97,87 @@ describe('grid connector - tree', () => {
     const childItems = [{ key: '1', name: 'foo bar' }];
     setChildItems(grid.$connector, rootItem, childItems);
 
-    await nextFrame();
     await nextFrame();
 
     // The column should have the expected width
     const expectedWidth = getBodyCellContent(grid, 0, 2)!.offsetWidth;
     const width = getBodyCellContent(grid, 0, 1)!.offsetWidth;
     expect(width).to.equal(expectedWidth);
   });
+
+  it('should request server for child items', async () => {
+    // Add an expanded root item
+    const rootItem = { key: '0', name: 'foo', children: true };
+    setRootItems(grid.$connector, [rootItem]);
+    expandItems(grid.$connector, [rootItem]);
+    await nextFrame();
+
+    // Wait for the request for child items to be sent
+    await aTimeout(GRID_CONNECTOR_PARENT_REQUEST_DELAY);
+    expect(grid.$server.setParentRequestedRanges).to.be.calledOnce;
+    expect(grid.$server.setParentRequestedRanges.firstCall.firstArg).to.deep.equal([{
+      parentKey: rootItem.key,
+      firstIndex: 0,
+      size: grid.pageSize
+    }]);
+  });
+
+  it('should render child nodes lazily', async () => {
+    // Add an expanded root item
+    const rootItem = { key: '0', name: 'foo', children: true };
+    setRootItems(grid.$connector, [rootItem]);
+    expandItems(grid.$connector, [rootItem]);
+    await nextFrame();
```

Comment:
I wonder if it would make more sense to use `await aTimeout(0);` here to make sure all the debouncers had a chance to run.